models/issue.go

```diff
@@ -729,6 +729,26 @@ func (issue *Issue) ChangeTitle(doer *User, oldTitle string) (err error) {
 	return sess.Commit()
 }
 
+// ChangeRef changes the branch of this issue, as the given user.
+func (issue *Issue) ChangeRef(doer *User, oldRef string) (err error) {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err = sess.Begin(); err != nil {
+		return err
+	}
+
+	if err = updateIssueCols(sess, issue, "ref"); err != nil {
+		return fmt.Errorf("updateIssueCols: %v", err)
+	}
+
+	if err = issue.loadRepo(sess); err != nil {
```

Comment:
Why is this needed here?

Reply:
I implemented this feature following a pattern of updating a title of an issue. ``updateIssueCols`` is needed to update 'ref' field in DB and save branch change while loadRepo retrieves this info from DB to make sure changes were indeed implemented which through API is returned to front-end (JS).

But writing this now, I realize that returning this info to JS might be unnecessary since the page is anyway reloaded once POST is completed.

Reply:
There's a tendency to avoid reloading the page if not required (so a half-typed comment will not be lost), but you don't need to concern about that in this PR unless you want to dig deeper.

Having said that, `loadRepo()` should not be called in a `models` function unless it's required for that function. If you happend to need that, you can call `LoadRepo()` from `services` or `routers` as you see fit.

Reply:
Oh I see, I'll remove it for this PR. But after loading the repo, I'd like to call ``createComment`` which I'm testing in another branch. And there, as far as I understand, I'll need it
![image](https://user-images.githubusercontent.com/20846919/69382936-8d28d280-0cb8-11ea-94fe-344b727002e6.png)